ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,254 @@
+
+            cloud_trace_spans.append(
+                {
+                    "name": span_name,
+                    "span_id": span_id,
+                    "display_name": get_truncatable_str_object(span.name),
+                    "start_time": start_time,
+                    "end_time": end_time,
+                    "parent_span_id": parent_id,
+                    "attributes": attributes,
+                    "links": _extract_links(span.links),
+                    "status": _extract_status(span.status),
+                    "time_events": _extract_events(span.events),
+                }
+            )
+
+        return cloud_trace_spans
+
+    def shutdown(self):
+        pass
+
+
+def _get_hexidecimal_trace_id(trace_id: int) -> str:
+    return "{:032x}".format(trace_id)
+
+
+def _get_hexidecimal_span_id(span_id: int) -> str:
+    return "{:016x}".format(span_id)
+
+
+def get_time_from_ns(nanoseconds):
+    """Given epoch nanoseconds, split into epoch milliseconds and remaining nanoseconds"""
+    if not nanoseconds:
+        return None
+    return {"seconds": int(nanoseconds / 1e9), "nanos": int(nanoseconds % 1e9)}
+
+
+def get_truncatable_str_object(str_to_convert, max_length=MAX_LENGTH):
+    """Truncate the string if it exceeds the length limit and record the truncated bytes
+        count.
+    """
+    truncated, truncated_byte_count = truncate_str(str_to_convert, max_length)
+
+    result = {
+        "value": truncated,
+        "truncated_byte_count": truncated_byte_count,
+    }
+    return result
+
+
+def truncate_str(str_to_check, limit=MAX_LENGTH):
+    """Check the length of a string. If exceeds limit, then truncate it.
+    """
+    str_bytes = str_to_check.encode("utf-8")
+    str_len = len(str_bytes)
+    truncated_byte_count = 0
+
+    if str_len > limit:
+        truncated_byte_count = str_len - limit
+        str_bytes = str_bytes[:limit]
+
+    result = str(str_bytes.decode("utf-8", errors="ignore"))
+
+    return result, truncated_byte_count
+
+
+def _extract_status(status: trace_api.Status):
+    """Convert a Status object to dict."""
+    if not status:
+        return None
+    status_json = {"details": None, "code": status.canonical_code.value}
+
+    if status.description is not None:
+        status_json["message"] = status.description
+
+    return status_json
+
+
+def _extract_links(links: Sequence[trace_api.Link]):
+    """Convert span.links"""
+    if not links:
+        return None
+    extracted_links = []
+    for link in links:
+        trace_id = _get_hexidecimal_trace_id(link.context.trace_id)
+        span_id = _get_hexidecimal_span_id(link.context.span_id)
+        extracted_links.append(
+            {
+                "trace_id": trace_id,
+                "span_id": span_id,
+                "type": "CHILD_LINKED_SPAN",
+                "attributes": _extract_attributes(link.attributes),
+            }
+        )
+    return {"link": extracted_links}
```

Comment:
It looks like this should also have a [`'dropped_links_count'` key](https://googleapis.dev/python/cloudtrace/latest/gapic/v2/types.html#google.cloud.trace_v2.types.Span.Links.dropped_links_count)? This will probably always be 0 in practice -- in which case it looks like it's fine to omit it -- since the SDK has a max of 32 links and stackdriver has a max of 128, but we may run into problems with custom SDKs later if we don't check the link count here.

It looks like the same is true for events below.